### ISSUE TITLE
Opening courses on Moodle 4.3 or 4.4

### ISSUE THREAD

**Issue by leehamgold42** [open-lms-open-source/moodle-theme_snap#286]
HI, I have use the snap theme for a long time now and I updated my moodle to 4.3 as I saw that there was an update, However when I go to access a course I get "
Fatal error: Declaration of theme_snap\output\core\course_renderer::course_section_cm_completion($course, $completioninfo, cm_info $mod, $displayoptions = []) must be compatible with core_course_renderer::course_section_cm_completion() in /home/hpscloud/public_html/moodle43/theme/snap/classes/output/core/course_renderer.php on line 292"

How can I resolved this? I have gone back to my previous moodle version of 4.1.1 with snap 4.1, and it's broken

**Comment by EdTechSavvy** [comment]
I am having the same problem with Snap theme PHP 8.1 Moodle version 4.3. 

Snap is using a method (core_course_renderer::course_section_cm_completion) that has been removed from core Moodle in 4.3+
From Moodle documentation:

=== 4.3 ===
* The core_course_renderer::course_section_cm_completion method has been removed, and can no longer be used
* External function core_course_external::get_course_contents() now returns a new field activitybadge with the data to display
  the activity badge when the module implements it.
* prepare_new_moduleinfo_data() now accepts a parameter "suffix" that will be added to the name of the completion rules.
* The method core_course_bulk_activity_completion_renderer:: edit_default_completion() has been deprecated and will be removed.
* The core_course\output\activity_information output class has been deprecated. Use core_course\output\activity_completion
  and core_course\output\activity_dates instead.
  
Alongside with that, the following methods have been deprecated too:
  - core_course_renderer::render_activity_information()
  - renderer_base::activity_information()

**Comment by leehamgold42** [comment]
@EdTechSavvy do you have any ideas of how to fix it because I have emailed OpenLMS about the issue, but nobody have gotten back to me or have released an update to the plugin.

**Comment by lvidgen** [comment]
Yes, this is broken and needs to be fixed by the dev team. Snap 4.3 will not work with Moodle 4.3 as reported in [Issue 284](https://github.com/open-lms-open-source/moodle-theme_snap/issues/284)

**Comment by peter-lernyx** [comment]
This has been pending apparently for 5 months now. Just as a guidance for the average admin, Is the dev team still planning to fix this or is the guidance to abandon Snap and move twds another template, since we we will have to move on from 4.1 at _some_ point?

**Comment by aolley** [comment]
> So can we assume that this is fixed for good?

It was fixed in this commit last month - so this specific issue - yes.
https://github.com/open-lms-open-source/moodle-theme_snap/commit/b3f96d9c5208f3c45edfeb2d96861af0ce4632b0

This is in the master, MOODLE_404_STABLE and MOODLE_403_STABLE branches.
